sentry-android-core/src/main/java/io/sentry/android/core/CurrentActivityHolder.java

```diff
@@ -0,0 +1,42 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.lang.ref.WeakReference;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public class CurrentActivityHolder {
+
+  private static @Nullable CurrentActivityHolder instance;
+
+  private @Nullable WeakReference<Activity> currentActivity;
+
+  public static CurrentActivityHolder getInstance() {
+    if (instance != null) {
+      return instance;
+    }
+    instance = new CurrentActivityHolder();
+    return instance;
+  }
```

Comment:
H: Check how other `getInstance()` is done, this implementation is not multithreaded safe.